src/boot/efi/console.c

```diff
@@ -193,6 +193,22 @@ EFI_STATUS console_set_mode(UINT32 mode) {
                         return change_mode(CONSOLE_MODE_FIRMWARE_RANGE_MIN);
                 return EFI_SUCCESS;
 
+        case CONSOLE_MODE_NEXT:
+                if (ST->ConOut->Mode->MaxMode <= CONSOLE_MODE_FIRMWARE_RANGE_MIN)
+                        return EFI_UNSUPPORTED;
+
+                /* If mode is invalid (-1) here we'll end up switching to mode 0, which is desired. */
```

Comment:
this is only true in two complement systems, and according to C spec such overflows are undefined behaviour which gcc famously uses for certain optimizations. hence, relying on two complement overflow is not safe in C. (other languages are not as nasty there).

i.e. the overflow should be handled explicitly, really.